multi_physics/QED/include/picsar_qed/containers/picsar_tables.hpp

```diff
@@ -895,6 +895,485 @@ namespace details{
+    * @tparam VectorType the vector type to be used (e.g. std::vector<double>)
+    * @tparam XMapFunctor a functor to map the index on the x axis to the corresponding coordinate x
+    * @tparam YMapFunctor a functor to map the index on the y axis to the corresponding coordinate y
+    * @tparam IXMapFunctor a functor to map the coordinate x to the corresponding index
+    * @tparam IYMapFunctor a functor to map the coordinate y to the corresponding index
+    */
+    template <
+        typename RealType, typename VectorType,
+        typename XMapFunctor, typename YMapFunctor,
+        typename IXMapFunctor, typename IYMapFunctor>
+    class generic_2d_table
+    {
+    public:
+
+        /**
+        * Constructor
+        *
+        * @param[in] how_many_x number of grid points along x
+        * @param[in] how_many_y number of grid points along y
+        * @param[in] values the values of the function that should be interpolated (row major order)
+        * @param[in] x_map_functor the functor mapping the index on the x axis to the corresponding coordinate x
+        * @param[in] y_map_functor the functor mapping the index on the y axis to the corresponding coordinate y
+        * @param[in] ix_map_functor the functor mapping the coordinate x to the corresponding index
+        * @param[in] iy_map_functor the functor mapping the coordinate y to the corresponding index
+        */
+        generic_2d_table(
+            int how_many_x, int how_many_y,
+            VectorType values,
+            XMapFunctor x_map_functor,
+            YMapFunctor y_map_functor,
+            IXMapFunctor ix_map_functor,
+            IYMapFunctor iy_map_functor):
+            m_how_many_x{how_many_x}, m_how_many_y{how_many_y},
+            m_values{values},
+            m_x_map_functor{x_map_functor}, m_y_map_functor{y_map_functor},
+            m_ix_map_functor{ix_map_functor}, m_iy_map_functor{iy_map_functor}
+            {}
+
+        /**
+        * Empty constructor
+        */
+        constexpr
+        generic_2d_table(){}
+
+        /**
+        * Constructor from byte array (not usable on GPUs)
+        *
+        * @param[in] raw_data a const reference to a byte vector
+        */
+        generic_2d_table(const std::vector<char>& raw_data)
+        {
+            using namespace picsar::multi_physics::utils;
+
+            constexpr size_t min_size =
+                sizeof(char)+//double or float
+                sizeof(m_how_many_x)+
+                sizeof(m_how_many_y);
+
+            if (raw_data.size() < min_size)
+                throw std::runtime_error("Binary data is too small \
+                to be a 2D table.");
+
+            auto it_raw_data = raw_data.begin();
+
+            if (serialization::get_out<char>(it_raw_data) !=
+                static_cast<char>(sizeof(RealType))){
+                throw std::runtime_error("Mismatch between RealType \
+                used to write and to read the 1D table");
```

Comment:
```suggestion
                used to write and to read the 2D table");
```
Looks like the same typo is also there line 463.

Reply:
thanks!